### ISSUE TITLE
.OS template variable is just "linux" on WSL instead of distro name

### ISSUE THREAD

**Issue by akoessler** [JanDeDobbeleer/oh-my-posh#1633]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

When using the .OS variable in the new templates, it is not possible to define different colors for different WSL distros.
The .OS variable only reacts to "linux".
Lines with concrete distro names like "ubuntu" or "WSL - ubuntu" are ignored.

With the example config below, this style is created:
![image](https://user-images.githubusercontent.com/11868357/150683871-a54e18fc-b075-4da8-a43c-3c9d553bf063.png)
Note the red background color, although the template says blue for "ubuntu".

### Desired behavior
The .OS variable in the template should be the distro name on WSL, or maybe "WSL - <distro-name>".
It should match the displayed name - everything else is just confusing.

### Additional note
Is there a way to see the values of all template variables? I did not find a way to do that, also not with the debug output.
And the .OS value did not match the displayed text.

### Theme

Custom theme to just repro this issue:
```
{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "final_space": true,
  "blocks": [
    {
      "type": "prompt",
      "alignment": "left",
      "segments": []
    },
    {
      "type": "prompt",
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "type": "os",
          "style": "powerline",
          "powerline_symbol": "\uE0B0",
          "foreground": "#FFFFFF",
          "background": "#555555",
          "background_templates": [
            "{{ if eq .OS \"ubuntu\" }}#0000FF{{ end }}",
            "{{ if eq .OS \"WSL - ubuntu\" }}#00FF00{{ end }}",
            "{{ if eq .OS \"linux\" }}#FF0000{{ end }}"
          ],
          "properties": {
            "display_distro_name": true
          }
        }
      ]
    }
  ]
}
```

### What OS are you seeing the problem on?

Windows

### Which shell are you using?

bash

### Log output

```shell
Version: 7.5.1

Segments:

ConsoleTitle(false) -   0 ms -
os(true)  -   0 ms -  WSL - ubuntu 

Run duration: 3.0232ms

Cache path: /home/akoessler

Logs:

2022/01/23 15:32:21 getArgs duration: 600ns, args:
2022/01/23 15:32:21 getArgs duration: 300ns, args:
2022/01/23 15:32:21 getShellName duration: 800ns, args:
2022/01/23 15:32:21 debug: getenv

2022/01/23 15:32:21 getenv duration: 9.9µs, args: OMP_CACHE_DISABLED
2022/01/23 15:32:21 isRunningAsRoot duration: 800ns, args:
2022/01/23 15:32:21 getShellName duration: 300ns, args:
2022/01/23 15:32:21 lastErrorCode duration: 400ns, args:
2022/01/23 15:32:21 debug: pwd
/mnt/c/Users/akoes
2022/01/23 15:32:21 pwd duration: 575.8µs, args:
2022/01/23 15:32:21 getPathSeperator duration: 300ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 200ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 200ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 200ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 300ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 300ns, args:
2022/01/23 15:32:21 getPathSeperator duration: 200ns, args:
2022/01/23 15:32:21 getCurrentUser duration: 500ns, args:
2022/01/23 15:32:21 getHostName duration: 1.3µs, args:
2022/01/23 15:32:21 getRuntimeGOOS duration: 200ns, args:
2022/01/23 15:32:21 templateCache duration: 623.1µs, args:
2022/01/23 15:32:21 debug: pwd
/mnt/c/Users/akoes
2022/01/23 15:32:21 pwd duration: 1.5µs, args:
2022/01/23 15:32:21 getRuntimeGOOS duration: 300ns, args:
2022/01/23 15:32:21 debug: getenv
Ubuntu
2022/01/23 15:32:21 getenv duration: 1.5µs, args: WSL_DISTRO_NAME
2022/01/23 15:32:21 isWsl duration: 4.2µs, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 500ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 500ns, args:
2022/01/23 15:32:21 templateCache duration: 500ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 templateCache duration: 300ns, args:
2022/01/23 15:32:21 templateCache duration: 500ns, args:
2022/01/23 15:32:21 templateCache duration: 400ns, args:
2022/01/23 15:32:21 debug: getenv

2022/01/23 15:32:21 getenv duration: 2.5µs, args: XDG_CACHE_HOME
2022/01/23 15:32:21 getCachePath duration: 18.5µs, args:
```

**Comment by JanDeDobbeleer** [comment]
This is a feature request and will follow in my new PR. I also just realized this.

**Comment by JanDeDobbeleer** [comment]
Will be handled by offering the `.WSL` boolean in templates.

**Comment by akoessler** [comment]
Will it be possible for the template to differentiate between "WSL - ubuntu" and "WSL - debian" with this `.WSL` boolean?

**Comment by JanDeDobbeleer** [comment]
@akoessler yes, that's the exact use-case.

**Comment by akoessler** [comment]
Can confirm this is resolved with 7.8.0 - thanks! ❤️
